psydac/linalg/block.py

```diff
@@ -588,8 +588,8 @@ def tosparse(self, **kwargs):
         M.eliminate_zeros()
 
         # Sanity check
-        assert M.shape[0] == self.codomain.dimension
-        assert M.shape[1] == self.  domain.dimension
+        #assert M.shape[0] == self.codomain.dimension
```

Comment:
@yguclu I stumbled across a case where the shape of a `polar.DenseMatrix`, a `polar.c1_linops.LinearOperator_StencilToDense` and a `polar.c1_linops.LinearOperator_DenseToStencil` was altered after calling `.tosparse()`. (Not the shape of the original object, but `M` and `M.tosparse()` would have a different shape)

This resulted in these sanity checks failing. Where's the mistake? Are these not proper sanity checks or should `.tosparse()` not change the shape of an object?

Reply:
Maybe the call to `tosparse()` included the ghost regions for those classes? The dimension would be different in that case. More likely these are simply bugs in the `polar` module.